File: core/src/main/java/com/tvsc/core/model/Episode.java
package com.tvsc.core.model;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.squareup.moshi.Json;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.experimental.Accessors;

import java.time.LocalDate;

/**
 * @author Taras Zubrei
 */
@Data
@NoArgsConstructor
@Accessors(chain = true)
public class Episode {
    private Long id;
    @JsonProperty("episodeName")
    @Json(name = "episodeName")
    private String name;
    @JsonProperty("airedEpisodeNumber")
    @Json(name = "airedEpisodeNumber")
    private Integer number;
    private LocalDate firstAired;
    private String overview;
    @JsonProperty("siteRating")
    @Json(name = "siteRating")
    private Double rating;
    private String imdbId;
    @JsonProperty("airedSeason")
    @Json(name = "airedSeason")
    private Integer season;
    private Boolean watched = false;
    @JsonProperty("filename")
    @Json(name = "filename")
    private String image;
}

File: core/src/main/kotlin/com/tvsc/core/model/Episode.kt
package com.tvsc.core.model

import com.fasterxml.jackson.annotation.JsonProperty
import java.time.LocalDate

/**
 * @author Taras Zubrei
 */
data class Episode (
        @JsonProperty                       var id: Long? = null,
        @JsonProperty("episodeName")        var name: String? = null,
        @JsonProperty("airedEpisodeNumber") var number: Int? = null,
        @JsonProperty                       var firstAired: LocalDate? = null,
        @JsonProperty                       var overview: String? = null,
        @JsonProperty("siteRating")         var rating: Double? = null,
        @JsonProperty                       var imdbId: String? = null,
        @JsonProperty("airedSeason")        var season: Int? = null,
                                            var watched: Boolean = false,
        @JsonProperty("filename")           var image: String? = null
        )
